src/libraries/System.Text.Json/tests/Common/JsonSchemaExporterTests.TestTypes.cs

```diff
@@ -0,0 +1,1320 @@
+                ExpectedJsonSchema: """
+                {
+                    "type": ["object","null"],
+                    "properties": {
+                        "Struct": {
+                            "type": "object",
+                            "properties": {
+                                "X": {"type":"integer"},
+                                "Y": {"type":"string"},
+                                "Z": {"type":"boolean"},
+                                "W": {"type":"number"}
+                            }
+                        },
+                        "NullableStruct": {
+                            "type": ["object","null"],
+                            "properties": {
+                                "X": {"type":"integer"},
+                                "Y": {"type":"string"},
+                                "Z": {"type":"boolean"},
+                                "W": {"type":"number"}
+                            }
+                        }
+                    }
+                }
+                """);
+
+            yield return new TestData<PocoWithNullableStructFollowedByStruct>(
+                Value: new() { NullableStruct = null, Struct = recordStruct },
+                AdditionalValues: [new() { NullableStruct = recordStruct, Struct = recordStruct }],
+                ExpectedJsonSchema: """
+                {
+                    "type": ["object","null"],
+                    "properties": {
+                        "NullableStruct": {
+                            "type": ["object","null"],
+                            "properties": {
+                                "X": {"type":"integer"},
+                                "Y": {"type":"string"},
+                                "Z": {"type":"boolean"},
+                                "W": {"type":"number"}
+                            }
+                        },
+                        "Struct": {
+                            "type": "object",
+                            "properties": {
+                                "X": {"type":"integer"},
+                                "Y": {"type":"string"},
+                                "Z": {"type":"boolean"},
+                                "W": {"type":"number"}
+                            }
+                        }
+                    }
+                }
+                """);
+
+            yield return new TestData<PocoWithExtensionDataProperty>(
+                Value: new() { Name = "name", ExtensionData = new() { ["x"] = 42 } },
+                ExpectedJsonSchema: """
+                    {
+                        "type": ["object","null"],
+                        "properties": {
+                            "Name": { "type": ["string", "null"] }
+                        }
+                    }
+                    """);
+
+            yield return new TestData<PocoDisallowingUnmappedMembers>(
```

Comment:
Would be good to see this for sealed types as well.

Reply:
The serializer doesn't disallow additional properties in sealed types. This can only be controlled via `UnmappedMemberHandling` setting.

Reply:
> The serializer doesn't disallow additional properties in sealed types.

Right. Wouldn't you want the schema to fail when the serialize does?

Reply:
I'm saying that the serializer doesn't fail in that case. Whether a type is sealed or not doesn't have any bearing on its serialization contract (since by default types don't support polymorphism).